app/scripts/directives/mobileServiceInstanceRow.js

```diff
@@ -0,0 +1,124 @@
+    row.serviceBindingsVersion = APIService.getPreferredVersion('servicebindings');
+    row.state = row.state || {};
+    var annotationSpace = "org.aerogear.binding." + this.apiObject.metadata.name;
+    var watches = [];
+    var bindingsWatch = false;
+    row.bindingsLimit = _.get(row.serviceClass, "spec.externalMetadata.bindingsLimit", 1);
+    _.extend(row, ListRowUtils.ui);
+
+    $scope.$on('$destroy', function(){
+        DataService.unwatchAll(watches);
+        DataService.unwatchAll([bindingsWatch]);
+    });
+
+    watches.push(DataService.watchObject(APIService.getPreferredVersion('mobileclients'), $routeParams.mobileclient, {namespace: $routeParams.project}, function(mobileClient) {
+        row.mobileClient = mobileClient;
+
+        row.bindingMeta = {
+          generateName: _.get(row, 'mobileClient.metadata.name').toLowerCase() + '-' + _.get(row, 'serviceClass.spec.externalMetadata.displayName').toLowerCase() + '-',
+          annotations: {
+            'binding.aerogear.org/consumer': _.get(row, 'mobileClient.metadata.name'),
+            'binding.aerogear.org/provider': _.get(row, 'apiObject.metadata.name')
+          }
+        };
+        
+        row.annotations = _.filter(row.mobileClient.metadata.annotations, function(annotation, name){
+          return name.split("/")[0] === annotationSpace;
+        })
+        .map(JSON.parse)
+        .sort(function(a, b) {
+          if (a.label < b.label) {
+            return -1;
+          }
+          if (a.label > b.label) {
+            return 1;
+          }
+          return 0;
+        });
+
+        row.servicePlanPromise = DataService.list(APIService.getPreferredVersion('clusterserviceplans'), {}, function(servicePlans) {
+          var servicePlanData = servicePlans.by('metadata.name');
+          row.providerServicePlan = _.find(servicePlanData, function(plan) {
+            return row.serviceClass.metadata.name === plan.spec.clusterServiceClassRef.name;
+          });
+          var kind = row.mobileClient.kind.toLowerCase();
+          var bindDataPath = 'providerServicePlan.spec.externalMetadata.' + kind + '_bind_parameters_data';
+          var bindData = _.get(row, bindDataPath, []);
+          row.bindData = _(bindData).map(JSON.parse).value();
+        });
+
+        if(!bindingsWatch){
+          bindingsWatch = DataService.watch(APIService.getPreferredVersion("servicebindings"), {namespace: $routeParams.project}, function(bindings){
+            row.bindings = _.filter(bindings.by("metadata.name"), function(binding){
+              return _.get(binding.metadata.annotations, "binding.aerogear.org/consumer") === row.mobileClient.metadata.name &&
+                    _.get(binding.metadata.annotations, "binding.aerogear.org/provider") === row.apiObject.metadata.name;
+            });
+      
+            row.bindingInProgress = false;
+            _.each(row.bindings, function(binding){
```

Comment:
Can this filter be moved to the template?

Reply:
I don't think so, as it needs to toggle off in any of the bindings in the array of bindings are not ready. the ng-if would only compare against one binding, wouldn't it?